resources/views/objects/search/_forum_options.blade.php

```diff
@@ -15,41 +15,62 @@
     You should have received a copy of the GNU Affero General Public License
     along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
 --}}
-<div class="search-advanced-forum-post">
-    <label class="search-advanced-forum-post__input-group">
-        <div class="search-advanced-forum-post__label">
-            {{ trans('home.search.forum_post.label.username') }}
-        </div>
-
-        <input
-            name="username"
-            value="{{ $search->urlParams()['username'] ?? '' }}"
-            class="search-advanced-forum-post__input search-advanced-forum-post__input--text"
-        >
-    </label>
-
-    @if (present($search->urlParams()['topic_id'] ?? null))
-        <label class="search-advanced-forum-post__input-group">
-            <div class="search-advanced-forum-post__label">
+@php
+// input field name mappings for view recycling.
+// pass in $fields to set; set a field to false to remove it.
+// TODO: hopefully this can be temporary ಠ_ಠ.
+$fieldDefaults = [
+    'forumId' => 'forum_id',
+    'topicId' => 'topic_id',
+    'user' => 'username',
+    'includeSubforums' => 'forum_children',
+];
+
+if (isset($fields)) {
+    $fields = array_merge($fieldDefaults, $fields);
+} else {
+    $fields = $fieldDefaults;
+}
+@endphp
+
+<div class="search-forum-options">
+    @if ($fields['user'] !== false)
+        <label class="search-forum-options__input-group">
+            <div class="search-forum-options__label">
+                {{ trans('home.search.forum_post.label.username') }}
+            </div>
+
+            <input
+                name="{{ $fields['user'] }}"
+                value="{{ request($fields['user']) }}"
+                class="search-forum-options__input search-forum-options__input--text"
+            >
+        </label>
+    @endif
+
+    {{-- FIXME: remove querystring check? --}}
+    @if ($fields['topicId'] !== false && present(request($fields['topicId'])))
```

Comment:
wtf is all this `!== false` check ಠ_ಠ

Reply:
I see ಠ_ಠ but why not just null

Reply:
`null` or `!isset` meant to use the existing value, until I changed it to always force the defaults, so I guess it has no meaning now 💃